There should be an EnityQualifier type of annotation to mark custom entity defining annotations (#644)
jakartaee/data: spec/src/main/asciidoc/jakarta-ee.adoc

- The `jakarta.nosql.Entity` annotation from the Jakarta NoSQL specification is an entity-defining annotation for Jakarta Data providers backed by NoSQL databases. Other Jakarta Data providers must not support the use of `jakarta.nosql.Entity` as an entity-defining annotation.

> 

Reviewer: probably-extraneous newline.

Author: Done: https://github.com/jakartaee/data/pull/644/commits/e37f4f5cfcbc6a9e4c396fb6a8774350d6a7c724

A Jakarta Data provider might define a custom entity-defining annotation.

> 
> To ensure interoperability within the Jakarta Data ecosystem, custom entity-defining annotations must be marked with `EntityQualifier`. This approach ensures that Jakarta Data providers can recognize and process entities declared with custom annotations, thereby maintaining consistency and flexibility in data management.

Reviewer: @otaviojava I know I'm being a bit annoying here, but t's enough to just state the requirements. It's a spec, not a whitepaper.

```suggestion
Custom entity-defining annotations must be marked with `@EntityQualifier`. 
This allows other Jakarta Data providers and tools to recognize and process entities declared with custom entity-defining annotations.
```

Author: Nope, you are not annoying.
I appreciate your patience.
